src/raw_node.rs

```diff
@@ -216,10 +216,14 @@ pub struct RawNode<T: Storage> {
     prev_hs: HardState,
 }
 
-impl<T: Storage> RawNode<T> {
+impl<'a, T: Storage> RawNode<T> {
```

Comment:
Do you want to tie this lifetime to the fn not the impl?

Reply:
Oh, yes, you're right! Thanks!